src/app/App.js

```diff
@@ -718,6 +719,19 @@ class App extends EventEmitter {
 		return false;
 	}
 
+	/**
+	 * An element with the autofocus attribute may gain focus before the DOMContentLoaded
+	 * event is fired. However, this event isn't triggered when navigating using Senna.
+	 * This is a wrapper that polyfills Browser's behavior.
+	 */
+	maybeTriggerFocusOnAutoFocusItems_(){
+		const autoFocusElement = globals.document.querySelector("[autoFocus]");
```

Comment:
The bad thing about going with this generic `autoFocus` is that in any tag that has it you can focus on,`div` if you add `tabindex`. So I get a little hesitant to go with this generic.

Reply:
Yeah, I can see this as a nice thing, but I don't think it will work in Portal. For tags, when `autoFocus` is passed, we simply do:

```javascript
<c:if test="<%= autoFocus %>">
	<aui:script>
		Liferay.Util.focusFormField('#<%= namespace + id %>');
	</aui:script>
</c:if>
```

So, unless we're trying to fix 2 different issues, we might simply make `focusFormField` aware of Senna, because we're not adding `autofocus` attributes anywhere.

Reply:
This utility doesn't make sense too much sense since we can use `autoFocus` attr and Senna gonna treat it.

Reply:
What if senna is disabled? What if we don't have access to the markup to add the `autoFocus` attr?

Reply:
I actually wasn't aware that [autofocus](https://caniuse.com/#feat=autofocus) was a native feature... maybe as you say, we should change our implementation in Liferay then... 🤔

Reply:
I think we should adopt `autoFocus` as it is native, when we no longer have Senna.js this will still work and other route libs will support, like `react-router`.

Reply:
Yeah, I'd agree... let's get this merged into both `2.x` and `3.x` and we can figure out the Portal side later